candidate_base.go

```diff
@@ -547,7 +548,10 @@ func UnmarshalCandidate(raw string) (Candidate, error) {
 				return nil, fmt.Errorf("%w: %v", errParsePort, parseErr) //nolint:errorlint
 			}
 			relatedPort = int(rawRelatedPort)
-		} else if split[0] == "tcptype" {
```

Comment:
Similarly, this parsing was also wrong. A candidate can both be relay (the above, default hidden by github) part of the if statement and declare a tcptype.

I've added an ice test case for this.